### PR TITLE
DEP: df.insert not showing warning for allow_dups and timezoneaware and naive indexing

### PR DESCRIPTION
**Pull request by phofl**
cc @jbrockmendel 

This ensures that the warning is raised also for allow_duplicates.

Additional question:

Should this:

```
index = pd.date_range("1/1/2013", "1/2/2013")
index.insert(0, Timestamp("2012-12-30", tz="UTC"))
```

raise the warning?

### CODE REVIEW

**Comment by jbrockmendel** [comment]
> Should this: [inserting tzaware Timestamp into tznaive DTI] raise the warning?

No.  It's behavior is not going to change.

**Comment by phofl** [comment]
In this case we could try avoiding the warning for insert completly, because its only raised because of column in self.columns and not because of the actual insert operation

File: pandas/tests/frame/indexing/test_insert.py [hunk 1]
```diff
@@ -104,3 +106,28 @@ def test_insert_frame(self):
...
+        # GH#44582
+        df = DataFrame(
+            [[1, 2], [3, 4]],
+            columns=date_range("1/1/2013", "1/2/2013"),
+        )
+
+        with tm.assert_produces_warning(FutureWarning):
+            df.insert(
+                loc=0,
+                column=Timestamp("2012-12-30", tz="UTC"),
```
Comment (jbrockmendel) [inline pandas/tests/frame/indexing/test_insert.py:121]: i thought the case you were interested in was the reverse: `column` being tznaive and `df.columns` being tzaware.  am i misunderstanding something?

Comment (phofl) [reply pandas/tests/frame/indexing/test_insert.py:121]: Does not really matter, both cases are showing the Warning, wenn allow duplicates is set to false

**Comment by jreback** [comment]
failures i think are legit on https://github.com/pandas-dev/pandas/runs/4315745835?check_suite_focus=true

File: pandas/tests/frame/indexing/test_insert.py [hunk 2]
```diff
@@ -104,3 +106,28 @@ def test_insert_frame(self):
...
+                value=0,
+                allow_duplicates=allow_duplicates,
+            )
+        expected = DataFrame(
+            [[0, 1, 2], [0, 3, 4]],
+            columns=[
+                Timestamp("2012-12-30", tz="UTC"),
+                Timestamp("2013-01-01", freq="1D"),
+                Timestamp("2013-01-02", freq="1D"),
```
Comment (jbrockmendel) [inline pandas/tests/frame/indexing/test_insert.py:130]: i think you're getting extraneous warnings breaking the npdev build bc of passing 'freq' here

File: pandas/tests/frame/indexing/test_insert.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/indexing/test_insert.py:121]: OK thanks

**Comment by jreback** [comment]
if you can rebase @phofl

**Comment by mroeschke** [comment]
Thanks for the PR, but it appears to have gone stale. Feel free to reopen if you have time to revisit.

**Pull request closed without merging**